opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/NoFieldQueryTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage.script.filter.lucene.relevance;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.opensearch.sql.data.model.ExprValueUtils;
+import org.opensearch.sql.expression.DSL;
+import org.opensearch.sql.expression.LiteralExpression;
+import org.opensearch.sql.expression.config.ExpressionConfig;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class NoFieldQueryTest {
+  NoFieldQuery query;
+  private final DSL dsl = new ExpressionConfig().dsl(new ExpressionConfig().functionRepository());
+  private final String testQueryName = "test_query";
+  private final Map<String, RelevanceQuery.QueryBuilderStep> actionMap
+      = ImmutableMap.of("paramA", (o, v) -> o);
+
+  @BeforeEach
+  void setUp() {
+    query = mock(NoFieldQuery.class,
+        Mockito.withSettings().useConstructor(actionMap)
+        .defaultAnswer(Mockito.CALLS_REAL_METHODS));
+    when(query.getQueryName()).thenReturn(testQueryName);
+  }
+
+  @Test
+  void createQueryBuilderTest() {
+    String sampleQuery = "field:query";
+
+    query.createQueryBuilder(List.of(
+        dsl.namedArgument("query",
+            new LiteralExpression(ExprValueUtils.stringValue(sampleQuery)))));
+
+    verify(query).createBuilder(eq(sampleQuery));
```

Comment:
I think there should also be another test for the other case where we call a function with noFields that is not `query`.

Reply:
Modified the exception message as the other case is not reachable since there are no other functions that extend NoFieldQuery.